db/kv_dense_block.go

```diff
@@ -594,7 +595,10 @@ func (d *DenseBlock) MakeLogEntry(indexEntry DenseBlockIndexEntry, entry DenseBl
 }
 
 func (d *DenseBlock) GetIndexEntry(position int64) (indexEntry DenseBlockIndexEntry) {
-	indexEntry = d.value[position : position+INDEX_ENTRY_LEN]
+	indexEntry, err := base.SafeSlice(d.value, int(position), int(position+INDEX_ENTRY_LEN))
+	if err != nil {
+		base.LogError(errors.New(fmt.Sprintf("Unable to GetIndexEntry from DensBlockIndexEntry, key: %v, cas: %v, startClock: %v, clock: %v",d.Key, d.cas, d.startClock, d.clock)))
```

Comment:
I think it's useful to include the Key here, and possibly the cas.  The clocks aren't going to be logged as anything useful, I don't think (just pointers), and I think it's critical to log the error text that's being returned from SafeSlice.